src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -40,7 +40,7 @@ def _query_account_key(cli_ctx, account_name):
 
 def _query_account_rg(cli_ctx, account_name):
     """Query the storage account's resource group, which the mgmt sdk requires."""
-    scf = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_STORAGE)
```

Comment:
Why not apply `storage_client_factory` for [_query_account_key](https://github.com/Azure/azure-cli/blob/2bc61d23b447585415b6d10611d36bc4c8c4f3a6/src/azure-cli/azure/cli/command_modules/storage/_validators.py#L29)? It looks `_query_account_rg` is used for listing storage accounts and `_query_account_key` is for listing keys.

Reply:
Good question. Sorry this change is not very relevant,  as I wanted to have one less way to create the storage client. That is it. I can undo this if you like